web-wallet/src/lib/components/Allocate/Allocate.svelte

```diff
@@ -80,9 +111,29 @@
   });
 
   $: fee = gasLimit * gasPrice;
-  $: isFromUnshielded = shieldedAmount > shieldedBalance;
-  $: isFromShielded = unshieldedAmount > unshieldedBalance;
-  $: isNextButtonDisabled = !(isFromUnshielded || isFromShielded);
+  $: difference = shielded - initialShielded;
+  $: isNextButtonDisabled =
+    difference === 0n || // No change in balance
+    shielded < 0n || // Shielded balance is negative
+    unshielded < 0n || // Unshielded balance is negative
+    shielded + unshielded > totalBalance ||
+    inputsInvalid;
+
+  function allocate() {
```

Comment:
See above about conventions.

More importantly this function should return a `Promise<string>` where the string is the hash of the completed transaction. The link to the explorer is broken otherwise.

Reply:
Thanks for pointing this out! Fixed!